src/v/iceberg/transaction.h

```diff
@@ -0,0 +1,67 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#pragma once
+
+#include "base/seastarx.h"
+#include "container/fragmented_vector.h"
+#include "iceberg/action.h"
+#include "iceberg/schema.h"
+#include "iceberg/table_metadata.h"
+#include "iceberg/table_requirement.h"
+#include "iceberg/table_update.h"
+
+#include <seastar/core/future.hh>
+
+#include <variant>
```

Comment:
this and several headers above like fragmented vector are reported unused

Reply:
Will fix in a follow-up